src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -373,73 +401,76 @@ public async Task StopAsync(CancellationToken cancellationToken)
 
         public async Task RestartHostAsync(CancellationToken cancellationToken)
         {
-            using (_metricsLogger.LatencyEvent(MetricEventNames.ScriptHostManagerRestartService))
+            if (!ShutdownRequested)
```

Comment:
nit: You could do the same thing as above and avoid nesting by returning if this is true